library/src/scripts/utility.ts

```diff
@@ -325,3 +325,28 @@ export function sanitizeUrl(url: string) {
         return "unsafe:" + url;
     }
 }
+
+/**
+ * Register an keyboard listener for the escape key.
+ *
+ * @param root - The element to watch for the escape listener in.
+ * @param returnElement - The element to return to when escape is pressed.
+ * @param callback
+ */
+export function addEscapeListener(
```

Comment:
This is thoroughly a DOM based utility method and should go into the `dom.ts` file as a result. `utility.ts` is for pure utility methods with no external dependencies outside of the javascript language.